airbyte-commons-worker/src/test/java/io/airbyte/workers/general/DefaultReplicationWorkerTest.java

```diff
@@ -231,14 +242,112 @@ void testReplicationRunnableSourceFailure() throws Exception {
         destination,
         messageTracker,
         recordSchemaValidator,
-        workerMetricReporter);
+        workerMetricReporter,
+        persistConfigHelper);
 
     final ReplicationOutput output = worker.run(syncInput, jobRoot);
     assertEquals(ReplicationStatus.FAILED, output.getReplicationAttemptSummary().getStatus());
     assertTrue(output.getFailures().stream()
         .anyMatch(f -> f.getFailureOrigin().equals(FailureOrigin.SOURCE) && f.getStacktrace().contains(SOURCE_ERROR_MESSAGE)));
   }
 
+  @Test
+  void testReplicationRunnableSourceUpdateConfig() throws Exception {
+    when(source.attemptRead()).thenReturn(Optional.of(RECORD_MESSAGE1), Optional.of(CONFIG_MESSAGE), Optional.empty());
+
+    final ReplicationWorker worker = new DefaultReplicationWorker(
+        JOB_ID,
+        JOB_ATTEMPT,
+        source,
+        mapper,
+        destination,
+        messageTracker,
+        recordSchemaValidator,
+        workerMetricReporter,
+        persistConfigHelper);
+
+    final ReplicationOutput output = worker.run(syncInput, jobRoot);
+    assertEquals(ReplicationStatus.COMPLETED, output.getReplicationAttemptSummary().getStatus());
+
+    verify(persistConfigHelper).persistSourceConfig(Long.valueOf(JOB_ID), CONNECTOR_CONFIG);
+  }
+
+  @Test
+  void testReplicationFailureOnSourceConfigPersistError() throws Exception {
+    when(source.attemptRead()).thenReturn(Optional.of(CONFIG_MESSAGE));
+    when(source.isFinished()).thenReturn(false, true);
+
+    final String PERSIST_ERROR_MESSAGE = "there was a problem persisting the new config";
+    doThrow(new ApiException(PERSIST_ERROR_MESSAGE)).when(persistConfigHelper).persistSourceConfig(Mockito.any(), Mockito.any());
+
+    final ReplicationWorker worker = new DefaultReplicationWorker(
+        JOB_ID,
+        JOB_ATTEMPT,
+        source,
+        mapper,
+        destination,
+        messageTracker,
+        recordSchemaValidator,
+        workerMetricReporter,
+        persistConfigHelper);
+
+    final ReplicationOutput output = worker.run(syncInput, jobRoot);
+    assertEquals(ReplicationStatus.FAILED, output.getReplicationAttemptSummary().getStatus());
```

Comment:
Note: we will fail the sync if there was an error when persisting the new config.